lib/node_manager.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:io';
+import 'dart:async';
+import "package:mime/mime.dart";
+
+abstract class NodeManager {
+  bool get isDataHost;
+
+  Future<ServerResponse> getRequest(ServerRequest sr);
+  Future<ServerResponse> deleteRequest(ServerRequest sr);
+  Future<ServerResponse> putRequest(ServerRequest sr, Map body);
+  Future<ServerResponse> postRequest(ServerRequest sr, dynamic body);
+  Future<Null> valueSubscribe(ServerRequest sr, WebSocket socket);
+}
+
+class ServerRequest {
+  String path;
+  HttpRequest request;
+  HttpResponse get response => request.response;
+  String get method => request.method;
+  bool isHtml = false;
+  bool get binary => request.uri.queryParameters.containsKey('binary');
```

Comment:
I'd keep the same naming convention here, either with `isXXX` or `hasXXX`, and personally I prefer `isBinary` over `binary`.